pixiedust/display/chart/renderers/mapbox/templates/iframesrcdoc.html

```diff
@@ -49,7 +49,11 @@
   }();
   {%endif%}
 </script>
-<iframe id="mapframe{{prefix}}" style='width:{{prefwidth}}px;height:{{prefheight}}px' srcdoc='{{body|oneline|replace("'","\\'")}}' onload="iframeLoaded(this)"></iframe>
+<iframe id="mapframe{{prefix}}" style='width:{{prefwidth}}px;height:{{prefheight}}px' srcdoc='{{body|btoa}}' onload="iframeLoaded(this)"></iframe>
+<script>
+  var iframe = document.getElementById('mapframe{{prefix}}');
+  iframe.setAttribute('srcdoc', atob('{{body|btoa}}'));
```

Comment:
@glynnbird Great idea to use base64 encoding. Can we make sure that body is not outputted twice in the generated markup? I suggest you use srcdoc="<html></html>" in the inline declaration and have the iframe.setAttribute('srcdoc', atob('{{body|btoa}}')); take care of updating the IFrame contents

Reply:
done. Thanks @DTAIEB.